### PR TITLE
[UR][Cuda][Hip] Add support for command-buffer kernel updates

### PR DESCRIPTION
**Pull request by fabiomestre**
Updates the call to urCommandBufferAppendKernelLaunchExp to use the new UR parameters.

Corresponding UR PR: https://github.com/oneapi-src/unified-runtime/pull/1924

### CODE REVIEW

File: sycl/source/detail/graph_impl.cpp [hunk 1]
```diff
@@ -1474,6 +1474,7 @@ void exec_graph_impl::updateImpl(std::shared_ptr<node_impl> Node) {
     }
   }
 
+  UpdateDesc.hNewKernel = UrKernel;
```
Comment (EwanC) [inline sycl/source/detail/graph_impl.cpp:1477]: Now that `hNewKernel` is optional, would `nullptr` be a better default until the SYCL-RT feature is implemented.

**Review by EwanC — APPROVED** [review]
LGTM, one minor question

**Review by aarongreig — APPROVED** [review]

**Comment by fabiomestre** [comment]
@intel/llvm-reviewers-runtime there was some tests failing on CI which are now fixed. This PR should be ready for review now.

**Comment by EwanC** [comment]
@intel/llvm-gatekeepers This is good to merge thanks

**Comment by nrspruit** [comment]
Please merge this, without this change any update of unified runtime fails to compile with latest intel/llvm due to the breaking abi change.

**Pull request merged**